component/loki/write/write.go

```diff
@@ -159,6 +160,13 @@ func (c *Component) Update(args component.Arguments) error {
 	}
 
 	cfgs := newArgs.convertClientConfigs()
+	uid := agentseed.Get().UID
+	for _, cfg := range cfgs {
+		if cfg.Headers == nil {
+			cfg.Headers = map[string]string{}
+		}
+		cfg.Headers["X-Agent-UID"] = uid
```

Comment:
nit: let's call it `X-Agent-ID`? (is is a unique id, but that's not relevant to the transport layer)